tests/test_warm_reboot.py

```diff
@@ -789,3 +795,674 @@ def test_swss_port_state_syncup(dvs):
             assert oper_status == "down"
         else:
             assert oper_status == "up"
+
+
+#############################################################################
+#                                                                           #
+#                        Routing Warm-Reboot Testing                        #
+#                                                                           #
+#############################################################################
+
+
+def clear_logs(dvs):
+    dvs.runcmd(['sh', '-c', '> /var/log/syslog'])
+    dvs.runcmd(['sh', '-c', '> /var/log/swss/swss.rec'])
+    dvs.runcmd(['sh', '-c', '> /var/log/swss/sairedis.rec'])
+
+
+def start_zebra(dvs):
+    dvs.runcmd(['sh', '-c', 'supervisorctl start zebra'])
+
+    # Let's give zebra a chance to connect to FPM.
+    time.sleep(5)
+
+
+def stop_zebra(dvs):
+    dvs.runcmd(['sh', '-c', 'pkill -x zebra'])
+    time.sleep(1)
+
+
+def start_fpmsyncd(dvs):
+    dvs.runcmd(['sh', '-c', 'supervisorctl start fpmsyncd'])
+
+    # Let's give zebra a chance to connect to Zebra.
+    time.sleep(5)
+
+
+def stop_fpmsyncd(dvs):
+    dvs.runcmd(['sh', '-c', 'pkill -x fpmsyncd'])
+    time.sleep(1)
+
+
+def enable_warmrestart(dvs, db, app_name):
+    create_entry_tbl(
+        db,
+        swsscommon.CFG_WARM_RESTART_TABLE_NAME, app_name,
+        [
+            ("enable", "true"),
+        ]
+    )
+
+
+def disable_warmrestart(dvs, db, app_name):
+    create_entry_tbl(
+        db,
+        swsscommon.CFG_WARM_RESTART_TABLE_NAME, app_name,
+        [
+            ("enable", "false"),
+        ]
+    )
+
+
+def set_restart_timer(dvs, db, app_name, value):
+    create_entry_tbl(
+        db,
+        swsscommon.CFG_WARM_RESTART_TABLE_NAME, app_name,
+        [
+            (app_name + "_timer", value),
```

Comment:
"config warm_restart bgp_timer" should be prepared as that for arp:

```
root@sonic:/home/admin# config warm_restart --help
Usage: config warm_restart [OPTIONS] COMMAND [ARGS]...

  warm_restart-related configuration tasks

Options:
  -s, --redis-unix-socket-path TEXT
                                  unix socket path for redis connection
  --help                          Show this message and exit.

Commands:
  disable
  enable
  neighsyncd_timer
root@sonic:/home/admin# 
```

Reply:
To avoid deferring this PR merge, i'll make this small adjustment right after i write this addition to sonic-utils. Then i'll come back to this UT code and will adjust accordingly.